caikit_nlp/resources/pretrained_model/hf_auto_causal_lm.py

```diff
@@ -168,6 +169,18 @@ def _get_data_collator(self, **kwargs) -> "transformers.DataCollator":
                 Collator to be used for causal language modeling.
         """
 
+        if "train_on_completion" in kwargs and kwargs["train_on_completion"]:
+            applicable_args = ["mlm", "response_template", "instruction_template"]
+            collator_kwargs = {
+                key: kwargs[key] for key in applicable_args if key in kwargs
+            }
+
+            if "mlm" not in collator_kwargs:
+                collator_kwargs["mlm"] = False
+
+            return DataCollatorForCompletionOnlyLM(
```

Comment:
The SFT Trainer hasn't been integrated quite yet, correct? Could you add a comment here for a TODO to validate that this can't be used if the trainer is initialized with `packing=True` so that we don't miss that edge case in the future? https://huggingface.co/docs/trl/v0.7.4/en/sft_trainer#train-on-completions-only

Reply:
background being we wnat to enable completion training for current PT and LOra which will be through thsi codebase